### PR TITLE
Bump workspace to rust edition 2018

### PR DESCRIPTION
**Pull request by wpaulino**
Mostly motivated by the need of async/await.

To reproduce, follow https://doc.rust-lang.org/cargo/commands/cargo-fix.html#edition-migration.

### CODE REVIEW

**Comment by dunxen** [comment]
Nice. Could you maybe list the combination of commands and args you used? Seems I keep missing some areas. Struggling to reproduce.

**Comment by dunxen** [comment]
> It was quite troublesome, I had to apply the migration multiple times due to numerous imports gated behind different features and config flags. I then committed the changes after each iteration and squashed them all into a single commit.

Yeah I think that's pretty much the trouble I'm dealing with. Not getting all the combos of config and features. I'll attempt again in the morning. Think I just had fuzzing left then maybe I can share a repro.

**Comment by tnull** [comment]
Seems this needs a rebase now?

**Comment by dunxen** [comment]
Alright, I've reproduced this but pretty much also lost track of all the steps. Phew!

File: lightning-invoice/src/de.rs [hunk 1]
```diff
@@ -715,15 +715,15 @@ impl From<ParseError> for ParseOrSemanticError {
...
+impl From<crate::SemanticError> for ParseOrSemanticError {
 	fn from(e: SemanticError) -> Self {
 		ParseOrSemanticError::SemanticError(e)
 	}
 }
 
 #[cfg(test)]
 mod test {
-	use de::ParseError;
+	use crate::de::ParseError;
```
Comment (tnull) [inline lightning-invoice/src/de.rs:726]: Could use `super` here and in quite a few places below.
```suggestion
	use super::ParseError;
```

**Review by tnull — COMMENTED** [review]
Looks good, now my finger hurts from scrolling.

File: lightning-invoice/src/de.rs (hunk 1 above)
Comment (tnull) [reply lightning-invoice/src/de.rs:726]: Yeah, fair enough.

**Comment by dunxen** [comment]
Oof. All those conflicts 😢

Basically looks good to me unless we still wanted something split out of here into a follow up, or are we happy with the whole package?

**Review by dunxen — APPROVED** [review]
Not sure if we wanted to split any of these changes out as discussed on discord, but this LGTM. Bring on the merge conflicts?

**Review by G8XSU — APPROVED** [review]

**Review by jkczyz — APPROVED** [review]
Everything looks crate!

**Comment by jkczyz** [comment]
> Everything looks crate!

Except the CI error... looks transient.

**Pull request merged**